app/controllers/application_controller.rb

```diff
@@ -3,30 +3,51 @@
 class ApplicationController < ActionController::Base
   include GDS::SSO::ControllerMethods
   before_action :set_request_id
-  before_action :authenticate_user!
-  before_action :set_user_instance_variable
+  before_action :authenticate
   before_action :check_service_unavailable
   default_form_builder GOVUKDesignSystemFormBuilder::FormBuilder
 
   before_action :clear_questions_session_data
 
-  def check_service_unavailable
-    if ENV["SERVICE_UNAVAILABLE"].present?
-      render "errors/service_unavailable", status: :service_unavailable, formats: :html
+  def authenticate
+    if Settings.basic_auth.enabled
+      basic_auth
+    else
+      signon_auth
+    end
+  end
+
+  def basic_auth
+    request.env["warden"].manager.config.intercept_401 = false
+    authenticate_or_request_with_http_basic do |name, password|
+      name == Settings.basic_auth.username && password == Settings.basic_auth.password
     end
+
+    @current_user = User.new(
+      name: Settings.basic_auth.username,
+      email: "#{Settings.basic_auth.username}@example.com",
+      organisation_slug: "government-digital-service",
+    )
   end
 
-  def set_user_instance_variable
+  def signon_auth
+    authenticate_user!
     @current_user = current_user
   end
 
+  def check_service_unavailable
+    if ENV["SERVICE_UNAVAILABLE"].present?
+      render "errors/service_unavailable", status: :service_unavailable, formats: :html
+    end
+  end
+
   def append_info_to_payload(payload)
     super
     payload[:host] = request.host
-    if current_user.present?
-      payload[:user_id] = current_user.id
-      payload[:user_email] = current_user.email
-      payload[:user_organisation_slug] = current_user.organisation_slug
+    if @current_user.present?
+      payload[:user_id] = @current_user.id
+      payload[:user_email] = @current_user.email
+      payload[:user_organisation_slug] = @current_user.organisation_slug
```

Comment:
This is why we need to be careful with using instance vars. Are we calling a `current_user` object or an instance var sort of confusion. The two could be very different

Reply:
The `ApplicationController` before this change was always setting the instance var `@current_user = current_user`. This won't work for basic auth because it doesn't effect the result of `current_user`. Therefore we set `@current_user` depending on the auth method and then make consistent use of it. The change here from `current_user` to `@current_user` is intentional.

The `HomeController` was calling the `current_user` method rather than the instance var which won't work because basic auth doesn't effect it. I think the change in this PR aligns on the use of the instance var which seems the Rail convention, especially for flows through controllers (HomeController is inheriting from the ApplicationController) and to their views. Personally I think using the instance var is easier to understand because it relies upon a variable we've set in our code and not the results of `current_user` which we have less visibility of how it's being set. Rails convention seems to be not to use instance vars for components because they're shared between views and instance var make less sense.

Happy to chat, but would like to keep the scope of this change limited to the auth strategies, greater refactoring of instance vars should be done separately I feel. I think this improves the consistent use of the `@current_user` v `current_user`.